packages/core/contracts-ovm/insured-bridge/implementation/BridgePoolFactory.sol

```diff
@@ -107,43 +118,97 @@ contract BridgePoolFactory is BridgePoolFactoryInterface, Ownable, OVM_CrossDoma
         emit SetDepositContract(depositContract);
     }
 
+    /**************************************************
+     *        CROSSDOMAIN ADMIN FUNCTIONS             *
+     **************************************************/
+
+    /**
+     * @notice Set this contract as the admin address in the L2 Deposit contract.
+     * @dev Only callable by the current owner.
+     * @param _l2Gas Gas limit to set for relayed message on L2
+     */
+    function setBridgeAdmin(uint32 _l2Gas) public onlyOwner depositContractSet {
+        sendCrossDomainMessage(
+            depositContract,
+            _l2Gas,
+            abi.encodeWithSignature("setBridgeAdmin(address)", address(this))
+        );
+        emit SetBridgeAdmin(address(this));
+    }
+
+    /**
+     * @notice Sets the minimum time between L2-->L1 token withdrawals in the L2 Deposit contract.
+     * @dev Only callable by the current owner.
+     * @param _minimumBridgingDelay the new minimum delay.
+     * @param _l2Gas Gas limit to set for relayed message on L2
+     */
+    function setMinimumBridgingDelay(uint64 _minimumBridgingDelay, uint32 _l2Gas) public onlyOwner depositContractSet {
+        sendCrossDomainMessage(
+            depositContract,
+            _l2Gas,
+            abi.encodeWithSignature("setMinimumBridgingDelay(uint64)", _minimumBridgingDelay)
+        );
+        emit SetMinimumBridgingDelay(_minimumBridgingDelay);
+    }
+
     /**
-     * @notice Privileged account can associate a whitelisted token with its linked token address on L2 and its
-     * BridgePool address on this network. The linked L2 token can thereafter be deposited into the Deposit contract
-     * on L2 and relayed via the BridgePool contract.
+     * @notice Owner can pause/unpause L2 deposits for all tokens.
+     * @dev Only callable by the current owner. Will set the same setting in the L2 Deposit contract via the cross
+     * domain messenger.
+     * @param _depositsEnabled bool to set if the deposit box should accept/reject deposits.
+     * @param _l2Gas Gas limit to set for relayed message on L2
+     */
+    function setEnableDeposits(bool _depositsEnabled, uint32 _l2Gas) public onlyOwner depositContractSet {
```

Comment:
this is a perfect reason why we can use the `toggle` method on L2 as the L1 contract (this contract) does not know if it's enabled or disabled. we _could_ store a bool in this contract that represents the enabled/disabled status of the L2 contract and only change it when this method is called and could then have that pattern? otherwise, are you ok with this setup?

Reply:
hmmm yes I think that's a good call. I can make that change.